### DISCUSSION TITLE
📚 Docs Wish List

### DISCUSSION THREAD

**Discussion by youmustfight** [xstate-wizards/xstate-wizards#74] [General]
What would you like to see in the documentation? Here's a running list of things that would be beneficial to include:
- how data assignments work with the state machine context object
- how "data models" appear on the state machine context object and are interacted with, for downstream database creates/updates/deletes
- the endpoints for session saving/fetching and the endpoint that takes the "data models" payload and interacts with an ORM

**Comment by mahmoud** [comment]
Oh nice, yes, I've been wondering all of the above!

If I can add others, I'd be interested in:

* **A broad-strokes conceptual intro to the abstractions introduced.** Clearly XState-Wizards is a higher-level wrapper around XState, so, assuming someone knows a thing or two about XState, what's the <5 minute pitch?
* **Quick reference on built-in events/actions.** - `CONTENT_NODE_BACK`, `SAVE_STATE`, etc. Maybe a Quick Start that only introduces these, and continues to build the most basic form possible?
* **More info about the dev and repo setup.** For instance, I notice `yarn` is enforced, I'm curious why (I've been using pnpm lately). I think the section on lerna and yalc could use some expansion, even if it's just linking to good entrypoints on how to use/debug them.
* **Some guidance on typing best practices would be great.** Despite XState being written in TypeScript, I noticed its docs on types and the typegen [[1]](https://xstate.js.org/docs/guides/typescript.html#typegen) features are still under active development. Just acknowledging that it may be a tall order given the somewhat dynamic state of the foundations.

**Comment by mahmoud** [comment]
@youmustfight thinking on it a bit more, I think even a screencast introduction to xstate-wizards could be a great "doc" at this point. Something that introduces the problem and the insights you captured in XSW's design. 

Should you opt for this, I'd recommend featuring an example, ideally one more minimal than any in the current examples directory. The react-wizards example is a pretty great comprehensive example, but something totally barebones just showing the structure would fit better.
